clients/python/girder_client/__init__.py

```diff
@@ -29,6 +29,24 @@
 import shutil
 import six
 import tempfile
+from six.moves.configparser import ConfigParser
+
+_config_defaults = {
+    'host': 'localhost',
+    'scheme':  'http',
+    'apiRoot': '/api/v1',
+    'apiUrl': None,
+    'port': None,
+    'apiKey': None,
+    'username': None,
+    'password': None,
+}
+CONFIG_DIR = os.environ.get(
+    'XDG_CONFIG_HOME', os.path.join(os.path.expanduser('~'), '.config'))
+config = ConfigParser(_config_defaults, allow_no_value=True)
+config.read([os.path.join(CONFIG_DIR, "girder-cli.conf")])
+if not config.has_section("girder_client"):
+    config.add_section("girder_client")
```

Comment:
All of this code should move inside `config.py` and should get wrapped in a function. We should not call that code at the time `girder_client` is imported; rather, it should be up to users to conditionally read the config file, and then pass that into the `GirderClient` constructor.